examples/md-flexible/src/configuration/YamlParser.cpp

```diff
@@ -1,386 +1,769 @@
-                                 it->second[config.massMap.name].as<double>());
-        }
-        continue;
+  // We iterate over all keys to identify known/unknown parameters.
+  for (auto itemIterator = node.begin(); itemIterator != node.end(); ++itemIterator) {
+    std::string key;
+    try {
+      key = itemIterator->first.as<std::string>();
+      m = node[key].Mark();
+
+      if (key == config.containerOptions.name) {
+        expected = "YAML-sequence of possible values.";
+        description = config.containerOptions.description;
+
+        config.containerOptions.value =
+            autopas::ContainerOption::parseOptions(autopas::utils::ArrayUtils::to_string(node[key], ", ", {"", ""}));
+
+        if (config.containerOptions.value.empty()) {
+          throw std::runtime_error("Parsed container list is empty. You used possibly an unknown container option.");
+        }
+
+      } else if (key == config.boxMin.name) {
+        expected = "YAML-sequence of three floats. Example: [0, 0, 0].";
+        description = config.boxMin.description;
+
+        auto tmpNode = node[key];
+        config.boxMin.value = {tmpNode[0].as<double>(), tmpNode[1].as<double>(), tmpNode[2].as<double>()};
+      } else if (key == config.boxMax.name) {
+        expected = "YAML-sequence of three floats. Example: [42, 42, 42].";
+        description = config.boxMax.description;
+
+        auto tmpNode = node[key];
+        config.boxMax.value = {tmpNode[0].as<double>(), tmpNode[1].as<double>(), tmpNode[2].as<double>()};
+      } else if (key == config.subdivideDimension.name) {
+        expected = "YAML-sequence of three ints in [0, 1].";
+        description = config.subdivideDimension.description;
+
+        auto tmpNode = node[key];
+        config.subdivideDimension.value = {tmpNode[0].as<bool>(), tmpNode[1].as<bool>(), tmpNode[2].as<bool>()};
+      } else if (key == config.loadBalancingInterval.name) {
+        expected = "Unsigned Integer";
+        description = config.loadBalancingInterval.description;
+
+        int tmp = node[key].as<int>();
+        if (tmp < 0) {
+          throw std::runtime_error("Load balancing interval must be a positive integer.");
+        }
+
+        config.loadBalancingInterval.value = tmp;
+      } else if (key == config.selectorStrategy.name) {
+        expected = "Exactly one selector strategy out of the possible values.";
+        description = config.selectorStrategy.description;
+
+        std::set<autopas::options::SelectorStrategyOption> parsedOptions;
+        if (node[key].IsSequence()) {
+          if (node[key].size() != 1) {
+            throw std::runtime_error("Pass Exactly one selector strategy.");
+          }
+          parsedOptions = autopas::SelectorStrategyOption::parseOptions(
+              autopas::utils::ArrayUtils::to_string(node[key], "", {"", ""}));
+        } else {
+          parsedOptions = autopas::SelectorStrategyOption::parseOptions(node[key].as<std::string>());
+        }
```

Comment:
```suggestion
        const auto optionAsString = [&]() {
            if (node[key].IsSequence()) {
                if (node[key].size() != 1) {
                    throw std::runtime_error("Pass Exactly one selector strategy.");
                }
                return autopas::utils::ArrayUtils::to_string(node[key], "", {"", ""}));
            } else {
                return node[key].as<std::string>());
            }
        }();
        const auto parsedOptions = autopas::SelectorStrategyOption::parseOptions(optionAsString);
```
a) only put the stuff that is actually part of the case differentiation into the if cases.
b) local lambda functions can help to compartmentalize logic, keep longer living stuff `const` and thus decrease the overall mental load.
c) This code is not tested at all so please check before applying 🙃

Reply:
I think this pattern occurs more than once.

Reply:
Wrote a extra function for this pattern, as it occurs multiples times.